### PR TITLE
moved compiler configuration into tsconfig.json

### PR DESCRIPTION
**Pull request by tomlarkworthy**
... so other tooling is configuration aware (e.g. vscode)
@mckoss

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -61,14 +61,6 @@ var TEST_SETS = [
 // Ignore ts-compile errors while watching (but not in normal builds).
 var watching = false;
 
-var TS_SETTINGS = {
-  sortOutput: true,
-  declarationFiles: true,
```
Comment (mckoss) [inline gulpfile.js]: Why did we get rid of settings for sortOutput, declarationFiles, and noExternalResolve?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "compilerOptions": {
+    "outDir": "lib",
+    "module": "commonjs",
+    "removeComments": true,
+    "preserveConstEnums": true,
+    "sourceMap": true,
+    "noEmitOnError": true
+  },
+  "files": [
+	"src/ast.ts",
```
Comment (mckoss) [inline tsconfig.json]: Wierd indentation?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	"src/bolt.ts",
+	"src/bolt.ts",
```
Comment (mckoss) [inline tsconfig.json]: bolt.ts listed twice?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "compilerOptions": {
+    "outDir": "lib",
+    "module": "commonjs",
+    "removeComments": true,
+    "preserveConstEnums": true,
+    "sourceMap": true,
+    "noEmitOnError": true
+  },
+  "files": [
```
Comment (mckoss) [inline tsconfig.json]: Can we not compile all *.ts files in src?

Comment (mckoss) [inline tsconfig.json]: and src/test?

**Comment by mckoss** [comment]
NOTE: Our Travis CI server is reporting success - BUT - there are 0 tests being executed - don't merge until fixed.

File: gulpfile.js (hunk 1 above)
Comment (tomlarkworthy) [inline gulpfile.js]: ASFAIK we don't save the .d.ts files so we don't need declarationFiles to be generated. 
The others are not actual compiler options.
sortFiles is used to preserve source-file maps across browserfy but it seems to works without this flag.
I am not sure why we had noExternalResolve becuase it also all seems to work without it. Do you know why we have them?

File: tsconfig.json (hunk 4 above)
Comment (tomlarkworthy) [inline tsconfig.json]: :) glob support in tsconfig is on the 1.8 roadmap!

**Comment by tomlarkworthy** [comment]
Yeah I fixed the issue and run-tests works properly now. it was generated lib/test/test for due to the TS output dir being counted twice.. I removed the compile-tests step altogether as things were getting compiled twice and we never really make a downstream distinction in the workflow anyway. It runs fast this way (and it simplifies the tsconfig with one compile set). 

PTAL @mckoss

**Pull request merged**